### PR TITLE
DTO validation

### LINKED ISSUE #14: Add validation to PollutedLocationDTO
## Description

Right now we don't have a good way to do DTO validation. Utilize [FluentValidation](https://docs.fluentvalidation.net/en/latest/) to achieve that.

## Steps to take

1. Create a `PollutedLocationDTO` validator (As a convention let's keep validators for DTOs in the same file)
2. Register validators in DI
3. Write some unit tests making sure that validation works

## Acceptance criteria

1. Validator for `PollutedLocationDTO`
2. Some unit tests written for checking validation

### PR DESCRIPTION
**Pull request by BenasB**
## What was done

- Added validators for `PollutedLocationDTO` and `LocationDTO` using `FluentValidation`
- Add unit tests checking valid and invalid DTOs
- Add validators to DI using `FluentValidation.DependencyInjectionExtensions`

### CODE REVIEW

File: Apsitvarkom.UnitTests/Models/Validation/PollutedLocationDTOValidationTests.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
+﻿using Apsitvarkom.Models.DTO;
+using FluentValidation;
+
+namespace Apsitvarkom.UnitTests.Models.Validation;
+
+public class PollutedLocationDTOValidationTests
+{
+    private static readonly IValidator<PollutedLocationDTO> _validator = new PollutedLocationDTOValidator();
```
Comment (6ar8nas) [inline Apsitvarkom.UnitTests/Models/Validation/PollutedLocationDTOValidationTests.cs]:
```suggestion
    private static readonly IValidator<PollutedLocationDTO> Validator = new PollutedLocationDTOValidator();
```

Consider using PascalCase for static variable

File: Apsitvarkom.UnitTests/Models/Validation/PollutedLocationDTOValidationTests.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            Progress = 0,
+            Radius = 0,
+            Severity = "Whatever",
+            Spotted = "timestamp"
+        },
+        new()
+        {
+            // Location (special case - not a primitive) fields are not initialized
+            Id = "hey",
+            Location = new(),
```
Comment (6ar8nas) [inline Apsitvarkom.UnitTests/Models/Validation/PollutedLocationDTOValidationTests.cs]:
```suggestion
            Location = new LocationDTO(),
```

I believe that would improve code readability

**Review by 6ar8nas — APPROVED** [review]
Good stuff. Noted a few minor issues to discuss.

File: Apsitvarkom.UnitTests/Models/Validation/PollutedLocationDTOValidationTests.cs (hunk 1 above)
Comment (BenasB) [reply Apsitvarkom.UnitTests/Models/Validation/PollutedLocationDTOValidationTests.cs]: ahh yeah my bad, this is basically a `const` for reference types, PascalCase preferred.

**Review by 6ar8nas — APPROVED** [review]

**Pull request merged**